tools/jdbc/src/jni/duckdb_java.cpp

```diff
@@ -310,13 +312,22 @@ static const char *const JDBC_STREAM_RESULTS = "jdbc_stream_results";
 jobject _duckdb_jdbc_startup(JNIEnv *env, jclass, jbyteArray database_j, jboolean read_only, jobject props) {
 	auto database = byte_array_to_string(env, database_j);
 	DBConfig config;
+	config.options.duckdb_api = "jni";
 	config.AddExtensionOption(
 	    JDBC_STREAM_RESULTS,
 	    "Whether to stream results. Only one ResultSet on a connection can be open at once when true",
 	    LogicalType::BOOLEAN);
 	if (read_only) {
 		config.options.access_mode = AccessMode::READ_ONLY;
 	}
+	const char *duckdb_api_key = "duckdb_api";
+	jstring jduckdb_api_key = env->NewStringUTF(duckdb_api_key);
+	jobject duckdb_api_value = env->CallObjectMethod(props, J_Map_remove, jduckdb_api_key);
+	if (duckdb_api_value) {
```

Comment:
Same here - we shouldn't need this if we add `duckdb_api` as a standard setting, no?

Reply:
The user could manually override it in that case during the setup of the `DBConfig` - but is that actually an issue? We are already allowing it to be overridden by the user at different layers anyway (e.g. here we are allowing it in the JDBC driver, we allow it in the C API, etc)?

Reply:
Another potential solution could be to make it so that in `DuckDBAPISetting::SetGlobal` we append to the current `duckdb_api` setting instead of replacing it. That might also be the desired behavior for `custom_user_agent`?

Reply:
Append is probably the safest, yeah

Reply:
made both settings appendable (`duckdb_api` is never empty, so a straightforward append, while `custom_user_agent` could be empty, so it gets either set or appended depending on current value).